src/main/java/org/opensearch/index/codec/customcodecs/Lucene95CustomStoredFieldsFormat.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.customcodecs;
+
+import org.apache.lucene.codecs.StoredFieldsFormat;
+import org.apache.lucene.codecs.StoredFieldsReader;
+import org.apache.lucene.codecs.StoredFieldsWriter;
+import org.apache.lucene.codecs.compressing.CompressionMode;
+import org.apache.lucene.codecs.lucene90.compressing.Lucene90CompressingStoredFieldsFormat;
+import org.apache.lucene.index.FieldInfos;
+import org.apache.lucene.index.SegmentInfo;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IOContext;
+import org.opensearch.index.codec.customcodecs.backward_codecs.Lucene95CustomCodec;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/** Stored field format used by pluggable codec */
+public class Lucene95CustomStoredFieldsFormat extends StoredFieldsFormat {
```

Comment:
This should go to `backward_codecs`, right?

Reply:
As of now, it won't possible because once we change the package of Lucene95CustomStoredFieldsFormat, ZstdCompressionMode and ZstdNoDictCompressionMode will not be accessible due to their access modifiers.

Reply:
This part I didn't get, sorry, which access modifiers? Those two are public ...

Reply:
oh yeah sorry, classes are public but their ctors are protected

for eg:
```
    protected ZstdCompressionMode() {
        this.compressionLevel = DEFAULT_COMPRESSION_LEVEL;
    }
```

Reply:
How does it relate to package placement?

Reply:
So this fails in compile time!

<img width="1258" alt="image" src="https://github.com/opensearch-project/custom-codecs/assets/25262500/d7e9727a-8d9c-43f0-8515-134adcb167d3">

The reason is mentioned over [here](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html)

> The protected modifier specifies that the member can only be accessed within its own package (as with package-private) and, in addition, by a subclass of its class in another package.

lmk if I am missing something

Reply:
Haha, indeed I forgot this subtle detail, please make it explicit:

```
        zstdCompressionMode = new ZstdCompressionMode(compressionLevel) {};
        zstdNoDictCompressionMode = new ZstdNoDictCompressionMode(compressionLevel) {};
``

Reply:
hahah, this totally didn't strike me. thank youu :D